### PR TITLE
[SYCL ] Test for redefining int specialization const to zero

### PR DESCRIPTION
**Pull request by maximdimakov**
This test is for https://github.com/intel/llvm/pull/4845
Signed-off-by: mdimakov <maxim.dimakov@intel.com>

### CODE REVIEW

**Comment by kbobrovs** [comment]
Jenkins failure fixed here: https://github.com/intel/llvm-test-suite/pull/542

File: SYCL/SpecConstants/2020/handler-api.cpp [hunk 1]
```diff
@@ -122,7 +122,7 @@ bool test_set_and_get_on_host(sycl::queue q) {
             "custom_type specializaiton constant before setting any value"))
       ++errors;
 
-    int new_int_value = 8;
+    int new_int_value = 0;
```
Comment (kbobrovs) [inline SYCL/SpecConstants/2020/handler-api.cpp]: I'd suggest to add new test cases rather than changing existing ones.

Comment (maximdimakov) [reply SYCL/SpecConstants/2020/handler-api.cpp]: I added new test case

**Review by AlexeySachkov — DISMISSED** [review]

**Review by AlexeySachkov — APPROVED** [review]

**Comment by maximdimakov** [comment]
@kbobrovs  could you, please, review?

**Review by kbobrovs — APPROVED** [review]
LGTM.

One note: it is not clear what this comment is about: `Remove XFAIL line due to problem is resolved in intel/llvm#4845`, as there is no such change in the PR or XFAIL line in the test.

**Pull request merged**